addons/stock_account/migrations/8.0.1.1/post-migrate.py

```diff
@@ -42,7 +44,65 @@ def create_properties(cr, pool):
                                context=ctx)
 
 
-@openupgrade.migrate()
+def propagate_invoice_state(cr):
+    """ Invoice state is now propagated from sale order to procurement to
+    stock move and picking. We trace it back from the picking and update
+    stock moves and procurements. """
+    openupgrade.logged_query(
+        cr,
+        """
+        UPDATE stock_move sm
```

Comment:
This was with only a couple of thousand move lines. But the performance should be inherent to the database layout. Comment done!